pkg/rekor/identity.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rekor
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+
+	"github.com/go-openapi/runtime"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+
+	hashedrekord_v001 "github.com/sigstore/rekor/pkg/types/hashedrekord/v0.0.1"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+)
+
+// IdentityEntry holds a certificate subject, issuer, and log entry metadata
+type IdentityEntry struct {
+	Subject string
+	Issuer  string
+	Index   int64
+	UUID    string
+}
+
+// Identities are the subjects/issuers to search
+type Identities struct {
+	Identities []Identity `json:"identities,omitempty"`
+}
+
+type Identity struct {
+	Subject string   `json:"subject"`
+	Issuers []string `json:"issuers,omitempty"`
+}
+
+// MatchedIndices returns a list of log indices that contain the requested identities.
+func MatchedIndices(logEntries []models.LogEntry, identities Identities) ([]IdentityEntry, error) {
+	if err := verifyIdentities(identities); err != nil {
+		return nil, err
+	}
+
+	var matchedIndices []IdentityEntry
+
+	for _, entries := range logEntries {
+		for uuid, entry := range entries {
+			entry := entry
+
+			certs, err := extractCertificates(&entry)
+			if err != nil {
```

Comment:
Do you want to add a log statement here like `Skipping entry xyz`, or do you want to differ between the errors that are unexpected (unmarshalling the entry type, decoding string) and the ones that are possible (other entry types)?

Reply:
I'm thinking to skip a log statement because the plan is to add support for all entry types and all key formats. It's just a limitation of the current tool. Also, given the number of entries we're ingesting, that'd be a lot of output.

Reply:
Ah, I guess with the verifier identities it should be easier to discern between expected and unexpected